### ISSUE TITLE
New contributor message

### ISSUE THREAD

**Issue by jrbourbeau** [dask/community#175]
It occurred to me that we could have a comment automatically posted when a new contributor opens their first issue and/or pull request. This would result in extra notifications, but would also allow us to provide additional information like:

- A friendly welcome message
- Link to https://blog.dask.org/2018/02/28/minimal-bug-reports
- Reference our code of conduct
- Ask contributors if they will need gpu-enabled CI, or include a message reminding a maintainer to enable gpuCI builds (xref https://github.com/dask/distributed/issues/5154) 
- ...

If there's interest in having some type of automated comments, a couple of ways we might consider implementing them would be:

- GitHub has its own [`actions/first-interaction` action](https://github.com/actions/first-interaction) for when users open up their first issue or pull request.
- Jupyter uses [this GitHub app](https://github.com/apps/welcome) to automatically post a comment when someone opens their first issue or pull request. [Here is an example](https://github.com/jupyterhub/repo2docker/pull/1062#issuecomment-877607252) of a pull request comment, and [here is the YAML file where the bot is configured](https://github.com/jupyterhub/.github/blob/9e1c9d7bec7129e3b0a28efcaef376af8298ae6f/.github/config.yml).

There are probably other options here too, these are just the two I'm aware of. 

Is there interest in have comments automatically posted when a new contributor opens their first issue or pull request? 

@quasiben @charlesbluca I'm curious if you have thoughts about including gpuCI-related information in such a comment

**Comment by quasiben** [comment]
I would be ok with this.  @jacobtomlinson might have thoughts here as well.  

As for gpuCI-related info, I have a draft PR up I am still working on we could link to

**Comment by GenevieveBuckley** [comment]
Is this intended to be for the benefit of the contributor, or the benefit of maintainers? Your four points seem to include a mix of things directed at different audiences.

> 1. A friendly welcome message
> 2. Link to https://blog.dask.org/2018/02/28/minimal-bug-reports
> 3. Reference our code of conduct
> 4. Ask contributors if they will need gpu-enabled CI, or include a message reminding a maintainer to enable gpuCI builds (xref [Stop GPU CI bot distributed#5154](https://github.com/dask/distributed/issues/5154))

1 seems vaguely nice. It might be of low-ish benefit to both contributors and maintainers (might be a more obvious nudge to be welcoming). Certainly whatever interaction contributors have with real people will have a much, much higher weight on how good or bad their experience is.

2 and 3 seem better placed in the issue template. It's a bit late if people see them only after the issue is submitted. Also, I imagine anyone who would ignore these points in the issue template would be just as likely to ignore them in an auto-generated comment, so I don't see how it helps.

4 seems like it's already covered by the gpu tester bot. Although it might be a good idea to improve the message that bot posts, since it's already confusing new-ish contributors a bit (I've been asked what the gpu bot comment was about and whether the contributor needed to do anything 

There is already a "first time contributor" badge, although that might only be shown for pull requests? I've noticed it occasionally on different repos, and I'm pretty sure it's the default & nothing special was done to set it up.

**Comment by charlesbluca** [comment]
> 2 and 3 seem better placed in the issue template. It's a bit late if people see them only after the issue is submitted. Also, I imagine anyone who would ignore these points in the issue template would be just as likely to ignore them in an auto-generated comment, so I don't see how it helps.

Agreed, I also feel like periodically updated information (such as the code of conduct, or information on how our tests work) is better suited in the templates so that it is exposed to long-time contributors that may have taken a gap from the project and want to be caught up, but would not get the first-time contributor message.

> it might be a good idea to improve the message that bot posts, since it's already confusing new-ish contributors a bit

I personally think that including a blurb in the Dask/Distributed PR templates explaining these comments (a condensed version of dask/dask#7985's content essentially) would go a long way in demystifying gpuCI for all contributors, although it is a realistic concern that contributors might just ignore this.

**Comment by jrbourbeau** [comment]
Thanks for the detailed responses @GenevieveBuckley @charlesbluca. My primary motivation here is to see if we could replace the gpuCI bot commenting on PRs from new contributors with some more generic bot comment that provided more useful information. There seems to be interest in including this information in issue/PR templates, which is totally fine, I'm happy to close this issue.

**Comment by jsignell** [comment]
Yeah I tend to be mildly anti general-purpose welcome bots. Especially if they are purporting to express a sentiment. I prefer bots to be to the point like the ones on conda-forge. Polite, but not effusive I guess.
